src/DIRAC/WorkloadManagementSystem/Service/SandboxStoreHandler.py

```diff
@@ -126,14 +130,35 @@ def transfer_fromClient(self, fileId, token, fileSize, fileHelper):
         else:
             hdPath = False
         # Write to local file
-        result = self.__networkToFile(fileHelper, hdPath)
+
+        if fileHelper:
+            result = self.__networkToFile(fileHelper, hdPath)
+        elif data:
+            hdPath = os.path.realpath(hdPath)
+            mkDir(os.path.dirname(hdPath))
+            with open(hdPath, "bw") as output:
+                output.write(data)
+            result = S_OK(hdPath)
+        else:
+            result = S_ERROR("No data provided")
+
         if not result["OK"]:
             gLogger.error("Error while receiving sandbox file", result["Message"])
             return result
         hdPath = result["Value"]
         gLogger.info("Wrote sandbox to file", hdPath)
         # Check hash!
-        if fileHelper.getHash() != aHash:
+        if fileHelper:
+            hdHash = fileHelper.getHash()
+        else:
+            oMD5 = hashlib.md5()
+            with open(hdPath, "rb") as fd:
+                bData = fd.read(10240)
```

Comment:
What is this number?

Reply:
This is the same chunk size as used in the SandboxStoreClient for the hash evaluation. I just keep it